internal/watcher/minio.go

```diff
@@ -81,42 +81,52 @@ func NewMinioWatcher(ctx context.Context, logger logr.Logger, config *MinioConfi
 	}, nil
 }
 
-func (w *minioWatcher) Watch(ctx context.Context) (*BlobEvent, error) {
-	event, err := w.pop(ctx)
+func (w *minioWatcher) Watch(ctx context.Context) (*BlobEvent, Cleanup, error) {
+	event, val, err := w.pop(ctx)
+	cleanup := w.rem(val)
```

Comment:
I think that `cleanup` should be declared after you're done checking the errors.

Reply:
okay that makes sense.